reflex-algos/src/main/python/mlcomp/parallelm/pipeline/dag.py

```diff
@@ -113,28 +112,40 @@ def sorted_execution_graph(self):
 
     def parent_data_objs(self, dag_node):
         data_objs = []
+        # dict to keep track of proper input index going to components
+        data_object_dict = {}
+        max_index = 0
         for parent in dag_node.parents():
             parent_id = parent[json_fields.PIPELINE_COMP_PARENTS_FIRST_FIELD]
             output_index = parent[json_fields.PIPELINE_COMP_PARENTS_SECOND_FIELD]
+            if json_fields.PIPELINE_COMP_PARENTS_THIRD_FIELD in parent:
```

Comment:
A more pythonic way (, which replaces the whole if else clause) is:
`input_index = parent.get(json_fields.PIPELINE_COMP_PARENTS_THIRD_FIELD, max_index)`